src/actions/move.ts

```diff
@@ -28,8 +29,16 @@ export class MoveActions {
     // Begin and end move.
     {
       name: 'Start move',
-      preconditionFn: (workspace) => this.mover.canMove(workspace),
-      callback: (workspace) => this.mover.startMove(workspace),
+      preconditionFn: (workspace) => {
+        const startBlock = this.getCurrentBlock(workspace);
+        return !!startBlock && this.mover.canMove(workspace, startBlock);
```

Comment:
Why not just `startBlock && ...`? '!!' seems unnecessary here. Ditto below in a few places.

Reply:
Needs to return a boolean, not null, and the compiler yelled.